### DISCUSSION TITLE
How to use "Grouping" in Dynamic Get ?

### DISCUSSION THREAD

**Discussion by sabrina974** [joomengine/Joomla-Component-Builder#719] [Q&A]
Hi, i use Joomla 3.9.25 and the last JCB.
I thought it's a little thing, but as i'm a beginner...
I wanted to do a Faq at my own, because, the questions-answers of JCb, doesn't convent for my own usage.

I have a categorie of question-answer.
And i have questions and answers.
I want to show all questions and answers grouping by category. That's easy.
But, in the front end, i have the categorie title repeat on each question and answer.
How to do, to show all category questions and answers, and in each category, a list of question and answer.

In my dynamic get, i use a "ListQuery", and i use "Grouping" on catid.
But, it shows me, one question-answer in each category, even i have multiple questions and answers 
in each category.
![Faq-remade](https://user-images.githubusercontent.com/4637575/113019840-4be4e480-9182-11eb-8db8-2011e3c5a376.jpg)
I hope, i've been clear.

Thanks for your answer,

Sabrina

**Comment by Llewellynvdm** [comment]
Hi @sabrina974 the question is clear, but how to achieve the desired outcome will take a few tries. I mean you must be willing to play around with a few different settings.

Now I know we achieved this kind of grouping in a few projects, but it always takes a few tries to find the correct outcome.

So looking at the dynamic get, the main get must retrieve the group, and the joined gets must be set to multiple and link on the group ID. That is the simple answer, but doing this you will end with a data set change, and you will have to change the way you use the data set in the view.

So the main get should look something like this:
![image](https://user-images.githubusercontent.com/5607939/209922827-d04a521a-0036-4055-8dc0-417526dd3a81.png)

Then the Join should look something like this:
![image](https://user-images.githubusercontent.com/5607939/209923068-97db5292-3f9c-4265-ac24-f5a8bd93d440.png)

This will build two queries and combine them into one final object. You will have to look at this object structure in the site-view to see how to access the data... but this is one of the many ways to achieve this. To be honest there is probably about four ways this can be done, with a dynamic get.

Give it a few tries and let me know if you not having success....
